features/user_login.feature

```diff
@@ -51,3 +51,13 @@ Feature: 'User logging'
     And I enter the mobile number
     And I enter the mobile number
     Then I should be allowed to login
+
```

Comment:
We should handle the case where a user is already logged in, but his role expires. In which case calling the REST api should return a `403`.

If the user has an **active** `Order fulfilment` role, but an **expired** `Order administrator`. He/she should be able to log in to stock, but should receive a `403` when calling an endpoint which is reserved to the Administrator.

We need to figure out those potential scenarios as well. @steveyken any other that come to mind ?

Reply:
Yes, that's a good summary. At it's core, we are checking the available permissions at the beginning of *each* request so the moment their role access expires, they loose the relevant abilities. 403 is sensible.

When a role expires, we may want to consider updating the app UI, so perhaps include the (read-only) role expiry date in the frontend models also (purely for UI reasons). We will want to think this through. It may also help in cases where the role expiry means the user no longer has access to the app entirely (e.g. can't login to Stock app), in which case, we will want to gracefully handle that. Perhaps forcing a logout and a screen saying "Goodbye and thank you for volunteering today", for example.